.github/actions/pre_cache/action.yml

```diff
@@ -0,0 +1,23 @@
+name: "Pre-Cache"
```

Comment:
Given that many of these actions are getting significantly simpler, I think it'd be worth considering just having this happen in an early stage of the `make_init` action. There was previously an argument for keeping this separate since it handled cache key generation for several different things, but at this point we're only creating a cache key for the venv.

Are there any cases where we'd want to run this action and `make_init` independently of each other? I can't think of any right now, so splitting the two up actually adds overhead given that you'll always have to call both back-to-back in workflows making use of them.

It's worth double-checking that there aren't any cases of `make_init` intentionally being run without caching currently in CircleCI before doing this, though.

Reply:
The only test that currently separates running `make_init` and `pre_cache` is the `python-prod-deps-smoke-test`

Reply:
Ah, I guess that one should have been obvious.

I wonder if, in that case, the thing that we'd want to do is have the action take a `dont_cache_venv` argument that's false by default but only set to true in the workflow that eventually replaces `python-prod-deps-smoke-test`

Reply:
That made more sense to me than keeping them separate at this point. I was able to refactor the `pre_cache` step into the `make_init` action with a `use_cached_venv` argument.